server/channels/store/searchlayer/user_layer.go

```diff
@@ -51,52 +52,56 @@ func (s *SearchUserStore) Search(teamId, term string, options *model.UserSearchO
 
 			usersIds, err := engine.SearchUsersInTeam(teamId, listOfAllowedChannels, sanitizedTerm, options)
 			if err != nil {
-				mlog.Warn("Encountered error on Search", mlog.String("search_engine", engine.GetName()), mlog.Err(err))
+				rctx.Logger().Warn("Encountered error on Search", mlog.String("search_engine", engine.GetName()), mlog.Err(err))
 				continue
 			}
 
 			users, nErr := s.UserStore.GetProfileByIds(context.Background(), usersIds, nil, false)
 			if nErr != nil {
-				mlog.Warn("Encountered error on Search", mlog.String("search_engine", engine.GetName()), mlog.Err(nErr))
+				rctx.Logger().Warn("Encountered error on Search", mlog.String("search_engine", engine.GetName()), mlog.Err(nErr))
 				continue
 			}
 
-			mlog.Debug("Using the first available search engine", mlog.String("search_engine", engine.GetName()))
+			rctx.Logger().Debug("Using the first available search engine", mlog.String("search_engine", engine.GetName()))
 			return users, nil
 		}
 	}
 
-	mlog.Debug("Using database search because no other search engine is available")
+	rctx.Logger().Debug("Using database search because no other search engine is available")
 
-	return s.UserStore.Search(teamId, term, options)
+	return s.UserStore.Search(rctx, teamId, term, options)
 }
 
-func (s *SearchUserStore) Update(user *model.User, trustedUpdateData bool) (*model.UserUpdate, error) {
-	userUpdate, err := s.UserStore.Update(user, trustedUpdateData)
+func (s *SearchUserStore) Update(rctx request.CTX, user *model.User, trustedUpdateData bool) (*model.UserUpdate, error) {
+	userUpdate, err := s.UserStore.Update(rctx, user, trustedUpdateData)
 
 	if err == nil {
-		s.rootStore.indexUser(userUpdate.New)
+		s.rootStore.indexUser(rctx, userUpdate.New)
 	}
 	return userUpdate, err
 }
 
 func (s *SearchUserStore) Save(user *model.User) (*model.User, error) {
+	//TODO: Use the actuall request context from the App layer
```

Comment:
Maybe we can add the jira ticket number to these

Reply:
Good idea, linked in 9361a87fb6